### PR TITLE
input: add missing windows multimedia keys

### PR DESCRIPTION
**Pull request by christoph-heinrich**
x11 and wayland had a lot of multimedia keys mapped that were missing on windows.
Now the only ones they map that windows doesn't are `MP_KEY_WWW`, `MP_KEY_ZOOMIN` and `MP_KEY_ZOOMOUT`, for which I couldn't  find any equivalent ones.

I have no idea if it's a good idea to map both virtual-key codes and appcommands for the same same functionality (e.g. mapping both `VK_PLAY` and `APPCOMMAND_MEDIA_PLAY` to `MP_KEY_PLAY`). I'm not a windows user so I couldn't test it out and I was unable to find an answer to that online.

I relied on the following for making those mappings:
https://cgit.freedesktop.org/xorg/proto/x11proto/tree/XF86keysym.h
https://learn.microsoft.com/en-us/windows/win32/inputdev/wm-appcommand
https://learn.microsoft.com/en-us/windows/win32/inputdev/virtual-key-codes

Completely untested

### CODE REVIEW

**Comment by kasper93** [comment]
> I have no idea if it's a good idea to map both virtual-key codes and appcommands for the same same functionality (e.g. mapping both VK_PLAY and APPCOMMAND_MEDIA_PLAY to MP_KEY_PLAY).

Nope, because you send MP_KEY_PLAY twice now.

Also things like `VK_VOLUME_*`  shouldn't be mapped. They are used to control global volume. And while you probably can hijack them, it is convenient to control system volume too. (with your patch, both system volume and mpv is adjusted with the same key)

`APPCOMMAND_*`  should be used to control volume, playback state, etc.

**Comment by christoph-heinrich** [comment]
> `APPCOMMAND_*` should be used to control volume, playback state, etc. Not `VK_` directly.

Thanks, I've removed all the overlapping `VK_*` ones.

File: osdep/w32_keyboard.c [hunk 1]
```diff
@@ -43,7 +43,7 @@ static const struct keymap vk_map_ext[] = {
 static const struct keymap vk_map[] = {
     // special keys
     {VK_ESCAPE, MP_KEY_ESC}, {VK_BACK, MP_KEY_BS}, {VK_TAB, MP_KEY_TAB},
-    {VK_RETURN, MP_KEY_ENTER}, {VK_PAUSE, MP_KEY_PAUSE},
```
Comment (Dudemanguy) [inline osdep/w32_keyboard.c:46]: I don't think you meant to remove `VK_PAUSE`?

Comment (kasper93) [reply osdep/w32_keyboard.c:46]: Why not? We have `APPCOMMAND_MEDIA_PAUSE` already.

Comment (Dudemanguy) [reply osdep/w32_keyboard.c:46]: Oh I didn't see that. So that magically will work the same I guess?

Comment (kasper93) [reply osdep/w32_keyboard.c:46]: Ok, I tested. `APPCOMMAND_MEDIA_PAUSE` is for dedicated "media pause" button. And `VK_PAUSE` is `Pause/Break` key. Not the same thing. So if we want `Pause/Break` key to send `MP_KEY_PAUSE` we should restore it as you said.

Also I'm really confused at this.
```
#PLAY cycle pause                       # toggle pause/playback mode
#PAUSE cycle pause                      # toggle pause/playback mode
#PLAYPAUSE cycle pause                  # toggle pause/playback mode
#PLAYONLY set pause no                  # unpause
#PAUSEONLY set pause yes                # pause
```
```
#define MP_KEY_PLAY             (MP_KEY_MM_BASE+2)
#define MP_KEY_PAUSE            (MP_KEY_MM_BASE+3)
#define MP_KEY_PLAYPAUSE        (MP_KEY_MM_BASE+4)
#define MP_KEY_PLAYONLY         (MP_KEY_MM_BASE+23)
#define MP_KEY_PAUSEONLY        (MP_KEY_MM_BASE+24)
```

Ok, not related to this PR... I might clean this up. Pause key should pause, not cycle. So at least our mappings should use `MP_KEY_PAUSEONLY ` or do breaking change and remove all this cycle bs from PLAY and PAUSE key.

Comment (Dudemanguy) [reply osdep/w32_keyboard.c:46]: Well I'm not sure if there's a reason to have `MP_KEY_PLAY` and `MP_KEY_PAUSE` in the first place. If we do keep them, then I don't see a problem with the current default mappings of cycle. But unrelated, yes.

**Review by Dudemanguy — APPROVED** [review]

**Pull request merged**